src/sqlfluff/api/simple.py

```diff
@@ -171,3 +173,40 @@ def parse(
     if parsed.tree is None:  # pragma: no cover
         return {}
     return parsed.tree.as_record(show_raw=True)
+
+def reflow_tree(segment: BaseSegment, last_inserted=None):
+    """Reflows the tree into one with consistent formatting."""
+    reflowed_segments = []
+    if segment.segments:
+        for segment in segment.segments:
+            if segment.is_code:
+                if segment.reflow_pre_space and (not last_inserted or last_inserted.type not in (WhitespaceSegment, NewlineSegment)):
+                    reflowed_segments.append(WhitespaceSegment())
+                    last_inserted = reflowed_segments[-1]
+                reflowed_segment, last_inserted = reflow_tree(segment, last_inserted=last_inserted)
+                reflowed_segments.append(reflowed_segment)
+        tuple(reflowed_segments)
+        return segment.__class__(tuple(reflowed_segments)), last_inserted
```

Comment:
This doesn't work for the same reasons that we have all the 'fix' functionality for the rules